cmd/acra-translator/server/server.go

```diff
@@ -115,30 +108,31 @@ func (server *ReaderServer) HandleConnectionString(parentContext context.Context
 
 	errCh := make(chan error)
 
-	listenerContext := server.listenerContext(parentContext)
-
 	// start accept new connections from connectionString
-	connectionChannel, err := common.AcceptConnections(listenerContext, connectionString, errCh)
+	connectionChannel, err := common.AcceptConnections(parentContext, listener, errCh)
 	if err != nil {
 		logger.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCantAcceptNewConnections).WithError(err).Errorf("Can't start to handle connection string %v", connectionString)
 		return err
 	}
 	// use to send close packets to all unclosed connections at end
-	server.waitGroup.Add(1)
+	server.backgroundWorkersSync.Add(1)
 	go func() {
-		defer server.waitGroup.Done()
+		defer server.backgroundWorkersSync.Done()
 		logger.WithField("connection_string", connectionString).Debugln("Start wrap new connections")
 		for {
 			var connection net.Conn
 			select {
 			case connection = <-connectionChannel:
 				break
 			case <-parentContext.Done():
-				logger.WithError(parentContext.Err()).Debugln("Stop wrapping new connections")
+				if !errors.As(parentContext.Err(), &context.Canceled) {
+					logger.WithError(parentContext.Err()).Debugln("Stop wrapping new connections")
+				}
```

Comment:
Doesn't `errors.As` assign to its second argument if the cast is successful? `context.Canceled` is global "constant variable" from the `context` package. Maybe you meant to use `errors.Is` here instead?

There are other places where `errors.As` is used similarly and may need attention.

Reply:
TBH, I didn't think too much on it. Now after reading descriptions of `Is` and `As` it seems to me that `As` may panic if it can't cast and `Is` will return false in such case, so `Is` is really more appropriate here.